src/control/cmd/daos_agent/security_rpc.go

```diff
@@ -52,41 +50,85 @@ type (
 
 	// SecurityModule is the security drpc module struct
 	SecurityModule struct {
-		log            logging.Logger
-		signCredential credSignerFn
-		credCache      *credentialCache
+		log              logging.Logger
+		signCredential   credSignerFn
+		credCache        *credentialCache
+		config           *securityConfig
+		validAuthMethods AuthValidSet
+	}
 
-		config *securityConfig
+	authArgs struct {
+		tag     auth.AuthTag
+		reqBody []byte
 	}
+
+	AuthValidSet map[auth.AuthTag]bool
 )
 
 var _ cache.ExpirableItem = (*cachedCredential)(nil)
 
+func getValidAuthMethods(cfg *securityConfig) (AuthValidSet, error) {
```

Comment:
There is a dangling question here: Why would a single DAOS system want to depend on multiple auth types? Unless there's a compelling use case for allowing this, it seems unnecessarily complex, and maybe easier for a user to mess up. I think the auth type needs to be a single type defined in the server config, as a central source. The agent can get this information from the servers.